radix-engine-stores/src/hash_tree/tree_store.rs

```diff
@@ -0,0 +1,163 @@
+use std::collections::HashMap;
+
+use radix_engine_interface::api::types::SubstateId;
+use radix_engine_interface::crypto::Hash;
+
+/// A nibble (4 LSBs of the contained u8).
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct Nib(pub u8);
+
+/// A sequence of nibbles (potentially not byte-aligned).
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct Nibs(pub Vec<Nib>);
+
+impl Nibs {
```

Comment:
Out of interest - why have we created all these types, rather than just using those from `types.rs`?

Reply:
The ones in `types.rs` look like they store data more efficiently (eg as a Vec<u8> rather than a Vec<Nib>) and would likely be more performant to use

Reply:
I think it's fine / sensible to change the Aptos code - let's just make sure that such changes (after the first commit) appear in separate commits, so we can properly track the changes.

I imagine this arbitrary second layer will introduce overhead and make things less performant.

Reply:
Perhaps for the Nodes, if we think we wish to separate them, perhaps let's maybe call them `RadixX` (for X the JMT name)?

Reply:
the naming is hard, I agree that `TreeNode` is just an arbitrary variation on `Node`, but is probably less ambiguous than `RadixNode` ;p

Instead of solving name clashes, I have another idea: if our plan is to freely evolve Aptos' code (that's how I understand https://github.com/radixdlt/radixdlt-scrypto/pull/763#discussion_r1084115736), then these 2 structs actually will be merged anyway.